From b8579bfe7bd73948b196980841a92c82fc1f9f5c Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 19 Jan 2024 09:20:02 +0530
Subject: [PATCH] arm64: dts: st: Add node for on-board hub connected on
 stm32mp257f-ev1

MP25-Eval board has on-board hub connected to the usbh-usb2phy1
interface

Change-Id: I75d7a083ea53d9bc4136c757cd1dfecb05398ddc
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370802
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 52d679c3b275..885b294dc4b5 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -769,6 +769,17 @@ &usb2_phy2 {
 &usbh {
 	status = "okay";
 
+	usbh_ehci: usb@482f0000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		/* onboard HUB */
+		hub@1 {
+			compatible = "usb424,2514";
+			reg = <1>;
+			vdd-supply = <&scmi_v3v3>;
+		};
+	};
+
 	usbh_ohci: usb@482e0000 {
 		status = "disabled";
 	};
-- 
2.39.5

